builtin/providers/vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -67,7 +67,7 @@ func TestAccVSphereVirtualMachine_basic(t *testing.T) {
 					resource.TestCheckResourceAttr(
 						"vsphere_virtual_machine.foo", "disk.#", "2"),
 					resource.TestCheckResourceAttr(
-						"vsphere_virtual_machine.foo", "disk.0.template", template),
+						"vsphere_virtual_machine.foo", "disk.3582676876.template", template),
```

Comment:
Maybe thats better suited for a different PR since design of the basic test is outside the scope of this PR.  
FYI:  The changes you see here where we reference "disk.1234123423" is the hash of the Set value.  Looking at other providers tests, I've realized the set hash value is reproducible given that your disk declaration is constant, which it is in the tests.  So I had to update each test that uses a disk to use a hash instead of an array index (ex. "disk.0")